### ISSUE TITLE
How to Improve YOLOv8 Fine-Tuning for Better Generalization Across Different Scenes

### ISSUE THREAD

**Issue by faezehprb** [ultralytics/ultralytics#18794]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm fine-tuning yolov8n to use it as a person detector for CCTV cameras. However, when I fine-tune the model using the commands below on data from multiple cameras in different scenes(6000 image), the model seems biased toward the training data. The performance of the fine-tuned model on unseen scenes is worse than the pre-trained yolov8n.pt.

I expected that fine-tuning the model would improve its performance, but instead, it has become biased.

What steps can I take to achieve a model with better generalization across different scenes?

### Additional

results = model.train(data="/content/drive/MyDrive/SuperVision/FineTuneYolo/data.yaml", epochs=1, device=0, imgsz=1280, patience=20)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @faezehprb, thank you for reaching out and sharing your query about fine-tuning YOLOv8 🚀!

We recommend checking out the [Ultralytics Docs](https://docs.ultralytics.com/) for detailed training and fine-tuning guidance. You may specifically want to visit our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) page, which includes suggestions to help improve training performance and generalization.

If this is a ❓ custom training question, please ensure you are providing as much context as possible. Details like dataset samples, data distribution across different scenes, and training logs would assist us greatly in understanding the issue. Additionally:
- Are all your scenes well-represented and balanced in your training data?
- Does your data follow the best practices detailed in our [Data Preparation Guide](https://docs.ultralytics.com/datasets/)?
- Have you experimented with advanced techniques such as data augmentation, transfer learning, or different hyperparameter settings?

If you're encountering unexpected 🐛 behavior, we kindly request a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) that demonstrates the issue.

To begin troubleshooting, ensure you're using the latest `ultralytics` package, as updates frequently resolve known issues. To upgrade, use the following command:

```bash
pip install -U ultralytics
```

## Environments

For optimal results, make sure you're running YOLO in one of the following verified environments, all of which come preloaded with the required dependencies:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

We invite you to join the Ultralytics community to discuss ideas and seek further assistance:
- Real-time chat? Join us on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions? Visit [Discourse](https://community.ultralytics.com/) 📝
- Share and learn on our [Subreddit](https://reddit.com/r/Ultralytics) 📚

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. These checks validate the correct operation of YOLO across various platforms and tasks.

This is an automated response to guide you in the right direction 😊. An Ultralytics engineer will follow up with additional assistance as soon as possible. Let us know if you have further details to share in the meantime! 🚀

**Comment by Y-T-G** [comment]
You can follow the suggestions here:

https://github.com/ultralytics/ultralytics/issues/18710?notification_referrer_id=NT_kwDOAetur7QxNDMwNjE0OTkwNzozMjIwNjUxMQ#issuecomment-2604061801

**Comment by faezehprb** [comment]
> You can follow the suggestions here:
> 
> [#18710](https://github.com/ultralytics/ultralytics/issues/18710)

Hi dear [Y-T-G](https://github.com/Y-T-G)
I really didn't understand the conclusion of https://github.com/ultralytics/ultralytics/issues/18710 for my work. can you help me more in my case?

I just want to detect person classes. The data.yaml file is like below:

```
names:
- 'Ped'
nc: 1
test: /home/mehrdad/Desktop/Detection/codes/finetune/Exp5/Dataset/val
train: /home/mehrdad/Desktop/Detection/codes/finetune/Exp5/Dataset/train
val: /home/mehrdad/Desktop/Detection/codes/finetune/Exp5/Dataset/val
```

and the code I'm using to find tune the yolov8n.pt model is like below:


```
from ultralytics import YOLO

model = YOLO('yolov8n.pt', 'detect')
results = model.train(data="/home/mehrdad/Desktop/Detection/codes/finetune/data.yaml", epochs=50, device=0, imgsz=1280, patience=5)
```

train logs : 

```
Ultralytics 8.3.39 🚀 Python-3.10.12 torch-2.5.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1080 Ti, 11162MiB)
engine/trainer: task=detect, mode=train, model=yolov8n.pt, data=/home/mehrdad/Desktop/SuperVision/Paryabi/codes/finetune/data.yaml, epochs=50, time=None, patience=5, batch=16, imgsz=1280, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
...
Transferred 58/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?3fa7e316-167e-4ea4-a0bb-74ec85e83494) or open in a [text editor](command:workbench.action.openLargeOutput?3fa7e316-167e-4ea4-a0bb-74ec85e83494). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...
train: Scanning /home/mehrdad/Desktop/SuperVision/Paryabi/codes/finetune/Exp5/Dataset/train/labels... 4805 images, 175 backgrounds, 0 corrupt: 100%|██████████| 4805/4805 [00:07<00:00, 631.44it/s]
train: WARNING ⚠️ /home/mehrdad/Desktop/SuperVision/Paryabi/codes/finetune/Exp5/Dataset/train/images/SynoClipVid5_frame_001815.jpg: 11 duplicate labels removed
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?653208cc-e1a6-430e-8155-4b9943f8ada3) or open in a [text editor](command:workbench.action.openLargeOutput?653208cc-e1a6-430e-8155-4b9943f8ada3). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

Plotting labels to runs/detect/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 1280 train, 1280 val
Using 8 dataloader workers
Logging results to runs/detect/train
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/50      8.29G      1.502      1.929       1.23         38       1280: 100%|██████████| 301/301 [03:22<00:00,  1.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:14<00:00,  2.64it/s]
                   all       1200       5492       0.92       0.86      0.935      0.638

... 

Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      41/50      7.93G     0.6898     0.3646      0.858         27       1280: 100%|██████████| 301/301 [03:18<00:00,  1.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.15it/s]
                   all       1200       5492      0.962      0.945       0.98      0.806


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      42/50      7.96G     0.6854     0.3623     0.8569         27       1280: 100%|██████████| 301/301 [03:17<00:00,  1.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.17it/s]
                   all       1200       5492      0.964      0.941      0.981      0.805

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      43/50      7.96G     0.6757     0.3573     0.8546         23       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:11<00:00,  3.17it/s]
                   all       1200       5492      0.962      0.949      0.982      0.812


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      44/50      7.95G     0.6703     0.3519     0.8535         13       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.16it/s]
                   all       1200       5492      0.966      0.945      0.982       0.81


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      45/50      7.93G     0.6656     0.3503     0.8516         26       1280: 100%|██████████| 301/301 [03:17<00:00,  1.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.15it/s]
                   all       1200       5492      0.965      0.946      0.982       0.81


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      46/50      7.96G     0.6631     0.3488     0.8514         20       1280: 100%|██████████| 301/301 [03:17<00:00,  1.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.13it/s]
                   all       1200       5492      0.966      0.945      0.982       0.81


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      47/50      7.96G     0.6519     0.3426     0.8486         16       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.11it/s]
                   all       1200       5492      0.964      0.948      0.983      0.813


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      48/50      7.95G     0.6519     0.3408     0.8484         30       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.13it/s]
                   all       1200       5492      0.965      0.947      0.983      0.814


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      49/50      7.93G     0.6489     0.3394     0.8468         29       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.16it/s]
                   all       1200       5492      0.962      0.948      0.983      0.812

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      50/50      7.96G     0.6435     0.3366     0.8454         29       1280: 100%|██████████| 301/301 [03:17<00:00,  1.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:12<00:00,  3.11it/s]
                   all       1200       5492      0.964      0.947      0.983      0.815


50 epochs completed in 2.959 hours.
Optimizer stripped from runs/detect/train/weights/last.pt, 6.4MB
Optimizer stripped from runs/detect/train/weights/best.pt, 6.4MB

Validating runs/detect/train/weights/best.pt...
Ultralytics 8.3.39 🚀 Python-3.10.12 torch-2.5.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1080 Ti, 11162MiB)
Model summary (fused): 168 layers, 3,005,843 parameters, 0 gradients, 8.1 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 38/38 [00:14<00:00,  2.58it/s]
                   all       1200       5492      0.965      0.947      0.983      0.815
Speed: 0.4ms preprocess, 6.2ms inference, 0.0ms loss, 0.8ms postprocess per image
Results saved to runs/detect/train

```
shall I set hyperparameters to other values or using layer freezing which I don't know how to do it to preserve yolov8n performance and improve the generalization of it on other scene which are not seen during the train process?

Any insights or best practices for optimizing fine-tuning in my case would be greatly appreciated!

Thanks in advance for your help. 🙏

**Comment by Y-T-G** [comment]
You should follow the suggestions in the linked issue which says to keep the number of classes the same as original pretrained model, and use lower learning rate and freezing.
